test/libgit2.jl

```diff
@@ -516,6 +516,11 @@ mktempdir() do dir
                 @test LibGit2.isdiff(repo, "HEAD")
                 @test LibGit2.isdirty(repo, cached=true)
                 @test LibGit2.isdiff(repo, "HEAD", cached=true)
+                diff = LibGit2.diff_tree(repo, LibGit2.GitTree(repo, "HEAD^{tree}"), "", cached=true)
+                diff_strs = split(sprint(show, diff[1]), '\n')
+                @test diff_strs[1] == "DiffDelta:"
+                @test diff_strs[2] == "Status: DELTA_MODIFIED"
+                @test diff_strs[3] == "Number of files: 2"
```

Comment:
Well, it's implicitly tested here, but yes I can add some checks for the `DiffFile` part of this output.